### DISCUSSION TITLE
Enabeling 1.21 Experimental Features

### DISCUSSION THREAD

**Discussion by ddattee** [itzg/docker-minecraft-server#2911] [Q&A] [answered]
Hi I'm running a 1.20.4 Fabric server and I would like to enable 1.21 experimental features while waiting for all my mods to be compatible with 1.21 and update to it later.
I tried editing manually my world's `level.dat` as it is the usual way to do it, but it seems it's replaced on every server restart so it doesn't seem to work.
I've looked at the whole documentation searching for options or ways to do this but I didn't find anything.

Any idea on how I can enable them and still use these images and automation (which are so practicle btw) ?

**Comment by ddattee** [comment]
A small correction : it seems the `level.dat` modification are kept as I downloaded current file from the server and it seems to have everything enabled but I don't see any experimental features client side still.
Still diging.

**Comment by ddattee** [comment, marked as answer]
All good in fact, once I added the required keys to `level.dat` the server actually reuse this file so it kept my modifications.
So all is good you can modify it and have your modification active.

**Reply by josephdpurcell** [reply]
@ddattee What were the "required keys" you added to `level.dat`? I've tired https://old.reddit.com/r/Minecraft/comments/11l5c4k/multi_server_with_120_additions/jbbxao3/ with no luck.

**Reply by josephdpurcell** [reply]
Also, did you run into https://github.com/itzg/docker-minecraft-bedrock-server/discussions/274#discussioncomment-4221622 where the server no longer shows up under LAN?

**Reply by ddattee** [reply]
> @ddattee What were the "required keys" you added to `level.dat`? I've tired https://old.reddit.com/r/Minecraft/comments/11l5c4k/multi_server_with_120_additions/jbbxao3/ with no luck.

I enabled those features : vanilla, update_1_21, bundle, trade_rebalance.
![image](https://github.com/user-attachments/assets/e636cb7c-8103-48fb-8452-6c8c3a27a5f7)

Not sure bundle and trade worked as I didn't check but I got the crafter all right and copper v1 (the expensive one) accessible in my world.
I did it manually using [NBTExplorer](https://github.com/jaquadro/NBTExplorer).
Then stoped the server, backed up the original `level.dat`, put my new one in place, restarted the container and it used the new one with my modifications so features were enabled.

> Also, did you run into [itzg/docker-minecraft-bedrock-server#274 (reply in thread)](https://github.com/itzg/docker-minecraft-bedrock-server/discussions/274#discussioncomment-4221622) where the server no longer shows up under LAN?

I'm on JAVA server with JAVA players so no bedrock issue ^^.

**Reply by josephdpurcell** [reply]
Thank you! This helped me confirm what I figured out for Bedrock https://github.com/itzg/docker-minecraft-bedrock-server/discussions/373#discussioncomment-10210491 which was ultimately to figure out [How to add WorldEdit to Minecraft Bedrock Server](https://github.com/itzg/docker-minecraft-bedrock-server/discussions/442).
